### PR TITLE
Implemented method QgsMssqlProvider::defaultValue

### PR DESCRIPTION
**Pull request by domi4484**
## Description

Implemented missing method QgsMssqlProvider::defaultValue

### CODE REVIEW

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 1]
```diff
@@ -473,7 +473,44 @@ QString QgsMssqlProvider::quotedIdentifier( const QString &value )
 
 QString QgsMssqlProvider::defaultValueClause( int fieldId ) const
 {
-  return mDefaultValues.value( fieldId, QString() );
+  QString defVal = mDefaultValues.value( fieldId, QString() );
```
Comment (elpaso) [inline src/providers/mssql/qgsmssqlprovider.cpp:476]:
```suggestion
  const QString defVal = mDefaultValues.value( fieldId, QString() );
```

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 2]
```diff
@@ -473,7 +473,44 @@ QString QgsMssqlProvider::quotedIdentifier( const QString &value )
...
+  {
+    return defVal;
+  }
+
+  return QString();
+}
+
+QVariant QgsMssqlProvider::defaultValue( int fieldId ) const
+{
+  QString defVal = mDefaultValues.value( fieldId, QString() );
```
Comment (elpaso) [inline src/providers/mssql/qgsmssqlprovider.cpp:490]:
```suggestion
  const QString defVal = mDefaultValues.value( fieldId, QString() );
```

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 3]
```diff
@@ -473,7 +473,44 @@ QString QgsMssqlProvider::quotedIdentifier( const QString &value )
...
+
+  if ( providerProperty( EvaluateDefaultValues, false ).toBool() && !defVal.isEmpty() )
+  {
+    QString sql = QStringLiteral( "select %1" )
```
Comment (elpaso) [inline src/providers/mssql/qgsmssqlprovider.cpp:494]:
```suggestion
    const QString sql = QStringLiteral( "select %1" )
```

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 4]
```diff
@@ -473,7 +473,44 @@ QString QgsMssqlProvider::quotedIdentifier( const QString &value )
...
+                  .arg( defVal );
+
+    QSqlQuery query = createQuery();
+    query.setForwardOnly( true );
+
+    if ( !query.exec( sql ) )
+    {
+      QgsDebugMsg( query.lastError().text() );
+      pushError( tr( "Could not execute query" ) );
```
Comment (elpaso) [inline src/providers/mssql/qgsmssqlprovider.cpp:503]: maybe better to also append the last error test here?

**Comment by elpaso** [comment]
Nice! Can you add a test for this?

**Comment by domi4484** [comment]
Thank you for your feedback @elpaso ! I will implement your suggestion and the test and come back for re-review.

**Comment by m-kuhn** [comment]
Thanks @domi4484 did you run the tests on your machine?

**Comment by domi4484** [comment]
@m-kuhn yes I run it locally.

**Comment by m-kuhn** [comment]
@elpaso does this look mergable to you?
Would be great if it had some time to cool down before the final release.

**Review by elpaso — APPROVED** [review]

**Pull request merged**